csharp/src/Ice/Communicator.cs

```diff
@@ -711,26 +726,24 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
                 }
                 _clientThreadPool = new IceInternal.ThreadPool(this, "Ice.ThreadPool.Client", 0);
 
-                //
                 // The default router/locator may have been set during the loading of plugins.
-                // Therefore we make sure it is not already set before checking the property.
-                //
-                if (GetDefaultRouter() == null)
+                // Therefore we only set it if it hasn't already been set.
+                try
                 {
-                    IRouterPrx? router = GetPropertyAsProxy("Ice.Default.Router", IRouterPrx.Factory);
-                    if (router != null)
-                    {
-                        SetDefaultRouter(router);
-                    }
+                    _defaultLocator ??= GetPropertyAsProxy("Ice.Default.Locator", ILocatorPrx.Factory);
+                }
+                catch(FormatException ex)
+                {
+                    throw new InvalidConfigurationException("invalid value for Ice.Default.Locator", ex);
                 }
 
-                if (GetDefaultLocator() == null)
+                try
                 {
-                    ILocatorPrx? locator = GetPropertyAsProxy("Ice.Default.Locator", ILocatorPrx.Factory);
-                    if (locator != null)
-                    {
-                        SetDefaultLocator(locator);
-                    }
+                    _defaultRouter ??= GetPropertyAsProxy("Ice.Default.Router", IRouterPrx.Factory);
+                }
+                catch(FormatException ex)
```

Comment:
missing space
```suggestion
                catch (FormatException ex)
```

Reply:
Force of habit... Hopefully it'll be easier to catch all the formatting warnings once your test-warning PR gets merged in!

Reply:
Visual Studio analyzers and .editorconfig, display those. maybe this doesn't work yet with VS code

Reply:
Or maybe I just don't have the fanciest C# plugin installed... I'll have to do some digging!